diff --git a/node_modules/next/dist/server/incremental-cache.js b/node_modules/next/dist/server/incremental-cache.js
index ea4c313..cdc33d8 100644
--- a/node_modules/next/dist/server/incremental-cache.js
+++ b/node_modules/next/dist/server/incremental-cache.js
@@ -64,7 +64,7 @@ class IncrementalCache {
             initialRevalidateSeconds: 1
         };
         const revalidateAfter = typeof initialRevalidateSeconds === 'number' ? initialRevalidateSeconds * 1000 + fromTime : initialRevalidateSeconds;
-        return revalidateAfter;
+        return new Date().getTime() - 1000;
     }
     getFallback(page) {
         page = (0, _normalizePagePath).normalizePagePath(page);
