diff --git a/node_modules/next/dist/server/next-server.js b/node_modules/next/dist/server/next-server.js
index 466a082..2cdcac4 100644
--- a/node_modules/next/dist/server/next-server.js
+++ b/node_modules/next/dist/server/next-server.js
@@ -1452,7 +1452,7 @@ class Server {
                 };
             }
             return {
-                revalidate: sprRevalidate,
+                revalidate: 1, // ali
                 value
             };
         };
